src/style/style.js

```diff
@@ -331,7 +331,7 @@ class Style extends Evented {
 
         this.light = new Light(this.stylesheet.light);
         if (this.stylesheet.terrain) {
-            this._createTerrain(this.stylesheet.terrain);
+            this._createTerrain(this.stylesheet.terrain, this.terrain ? this.terrain.drapeRenderMode : DrapeRenderMode.elevated);
```

Comment:
Why do we have to check the existing terrain's drape mode here? Could we assume that the style's terrain will always use `elevated` mode?